rules.mk

```diff
@@ -47,19 +50,24 @@ out/$T/$(1)/$(2)/cmdline out/$T/$(1)/$(2)/initrd.img out/$T/$(1)/$(2)/kernel: ou
 out/$T/$(1)/$(2)/hook.yaml: $$(LINUXKIT_CONFIG)
 	sed '/hook-\(bootkit\|docker\):/ { s|:latest|:$T-$(2)|; s|quay.io/tinkerbell|$(ORG)|; }' $$< > $$@
 	if [[ $(1) == dbg ]]; then
-	    sed -i '/^\s*dbg/ s|#dbg||' $$@
+	    sed -i '/^\s*#dbg/ s|#dbg||' $$@
 	fi
+
+out/$T/$(1)/hook_$(DOCKER_2_HW_$(2)).tar.gz: out/$T/$(1)/initramfs-$(DOCKER_2_HW_$(2)) out/$T/$(1)/vmlinuz-$(DOCKER_2_HW_$(2))
+out/$T/$(1)/initramfs-$(DOCKER_2_HW_$(2)): out/$T/$(1)/$(2)/initrd.img
+out/$T/$(1)/vmlinuz-$(DOCKER_2_HW_$(2)): out/$T/$(1)/$(2)/kernel
+dist-files += out/$T/$(1)/initramfs-$(DOCKER_2_HW_$(2)) out/$T/$(1)/vmlinuz-$(DOCKER_2_HW_$(2))
```

Comment:
I encourage you to use less make over more. this addition is difficult to understand.

Reply:
Not a fan of having make do unnecessary work but I hear ya, this was new to me too. So its gone now. I was able to give the macro parameters better names (something I tried before but didn't work out until I changed some things for other reasons). Can you take a look at the last commit and see if that makes things a little clearer?